File: src/Input.scala
import scala.util.Random

object Input {
  def rndSeq: Seq[Int] = seqFromSeed(Random.nextInt())

  def seqFromSeed(seed: Int): Seq[Int] = {
    val random = new Random(seed)
    Stream.continually(random.nextInt(100))
  }

  def seq(n: Int, seed: Int): Seq[Int] = seqFromSeed(seed).take(n)
}

File: src/javafoo/Input.java
package javafoo;

import java.util.Random;

public class Input {
    public final int[] a;

    public Input(int n, int seed) {
        a = new int[n];
        Random random = new Random(seed);
        for (int i = 0; i < a.length; i++) {
            a[i] = random.nextInt(100);
        }
    }
}
